src/source/Ice/SocketConnection.c

```diff
@@ -260,6 +260,7 @@ STATUS socketConnectionReadData(PSocketConnection pSocketConnection, PBYTE pBuf,
                     continueRead = FALSE;
                     break;
                 case SSL_ERROR_WANT_READ:
+                case SSL_ERROR_ZERO_RETURN:
```

Comment:
I think we should mark the connection as closed to let the above call knows this state update.

Reply:
Actually, nvm, the documentation apparently says that there might still be some pending data.